packages/metadata-common/src/MetadataWidget.tsx

```diff
@@ -0,0 +1,250 @@
+        >
+          <div id={metadata.name}>{this.renderExpandableContent(metadata)}</div>
+        </ExpandableComponent>
+      </div>
+    );
+  };
+
+  render(): React.ReactElement {
+    return (
+      <div>
+        <div id="elyra-metadata">
+          <div>{this.props.metadata.map(this.renderMetadata)}</div>
+        </div>
+      </div>
+    );
+  }
+}
+
+/**
+ * MetadataWidget props.
+ */
+export interface IMetadataWidgetProps {
+  app: JupyterFrontEnd;
+  display_name: string;
+  namespace: string;
+  schema: string;
+}
+
+/**
+ * A abstract widget for viewing metadata.
+ */
+export class MetadataWidget extends ReactWidget {
+  renderSignal: Signal<this, any>;
+  props: IMetadataWidgetProps;
+
+  constructor(props: IMetadataWidgetProps) {
+    super();
+    this.addClass('elyra-metadata');
+
+    this.props = props;
+    this.renderSignal = new Signal<this, any>(this);
+
+    this.fetchMetadata = this.fetchMetadata.bind(this);
+    this.updateMetadata = this.updateMetadata.bind(this);
+    this.openMetadataEditor = this.openMetadataEditor.bind(this);
+    this.renderDisplay = this.renderDisplay.bind(this);
+  }
+
+  addMetadata(): void {
+    this.openMetadataEditor({
+      onSave: this.updateMetadata,
+      namespace: this.props.namespace,
+      schema: this.props.schema
+    });
+  }
+
+  /**
+   * Request metadata from server and format it as expected by the
+   * instance of `MetadataDisplay` rendered in `renderDisplay`
+   *
+   * Classes that extend MetadataWidget should override this
+   *
+   * @returns metadata in the format expected by `renderDisplay`
+   */
+  async fetchMetadata(): Promise<any> {
+    return await FrontendServices.getMetadata(this.props.namespace);
+  }
+
+  updateMetadata(): void {
+    this.fetchMetadata().then((metadata: any[]) => {
+      this.renderSignal.emit(metadata);
+    });
+  }
+
+  // Triggered when the widget button on side panel is clicked
+  onAfterShow(msg: Message): void {
+    this.updateMetadata();
+  }
+
+  openMetadataEditor = (args: any): void => {
+    this.props.app.commands.execute(commands.OPEN_METADATA_EDITOR, args);
+  };
+
+  /**
+   * Classes that extend MetadataWidget should override this
+   *
+   * @returns a rendered instance of `MetadataDisplay`
+   */
+  renderDisplay(metadata: IMetadata[]): React.ReactElement {
+    return (
+      <MetadataDisplay
+        metadata={metadata}
+        updateMetadata={this.updateMetadata}
+        openMetadataEditor={this.openMetadataEditor}
+        namespace={this.props.namespace}
+        schema={this.props.schema}
+      />
+    );
+  }
+
+  render(): React.ReactElement {
+    return (
+      <div>
+        <header className={METADATA_HEADER_CLASS}>
+          <div style={{ display: 'flex' }}>
+            <p> {this.props.display_name} </p>
+          </div>
+          <button
+            className={METADATA_HEADER_BUTTON_CLASS}
+            onClick={this.addMetadata.bind(this)}
```

Comment:
button here should include a `title` attribute so a tooltip can appear on hover. e.g.,

<img width="162" alt="image" src="https://user-images.githubusercontent.com/13156555/88714854-5cbcf200-d0eb-11ea-866e-dd0be150e718.png">

Reply:
I noticed this while working on moving Code Snippet to extend Metadata. It's an "easy" add, but I'm unsure what to put in the text without adding a new part just for the text. I already have a param for the header text, but that is usually plural. I could alway just have it say "Create new" but I don't think that's very good. Any ideas?

Reply:
what about the title used in the form? or is that not accessible in the sidebar component:

<img width="405" alt="image" src="https://user-images.githubusercontent.com/13156555/88716865-b6262080-d0ed-11ea-9048-475f189c26b6.png">

Reply:
we can use that, but that isn't a title, it's just the schema name (which in Code Snippets case would be `code-snippet`) #734 is currently open to update that to use the new `display_name` value on a schema, but this code does't have access to the schema content, just the name. I'm hesitant to add an api call to grab the schema just to get a display name, but I'm not 100% opposed to it.

Reply:
Seems like we pound the server for all instances of metadata - not sure how often for schemas, but I would argue schemas should be fetched exactly once at startup and be available to all front-end components.  Since they're static and we don't have any widgets relative to user-defined schema, this should be fine.

Reply:
It sound like I should add another call to `getSchema` here to deal with this then, but with the expectation that we update that function to cache the response of previous calls. 

I would suggest we deal with that cacheing in the refactor purposed #785 since currently that function can "return" error dialogs, which would make caching responses a hassle. But that purposed refactor would remove that problem. If tabling that cache is ok with you @kevin-bates then I'll add it to my refactor proposal.

As for how often `getSchema` is currently called, it is called each time a MetadataEditor is opened currently. This would add one addition call when a MetadataWidget is created (so based on our current code, once on startup)

Reply:
Deferring the refactor is fine - thanks!